src/scripts/main.js

```diff
@@ -1,3 +1,25 @@
 'use strict';
 
-// write code here
+const list = document.querySelector('.tree');
+
+list.addEventListener('click', (e) => {
+  const element = e.target.closest('li');
+
+  if (element) {
+    Array.from(element.children).forEach((child) => {
+      if (child.tagName !== 'SPAN') {
+        child.style.display = child.style.display === 'none' ? '' : 'none';
+      }
+    });
+  }
+});
+
+document.body.addEventListener('click', (e) => {
+  if (!list.contains(e.target)) {
+    const allElements = document.querySelectorAll('.tree li ul');
+
+    allElements.forEach((ul) => {
+      ul.style.display = 'none';
+    });
+  }
+});
```

Comment:
This event listener on the document body will close all tree elements when clicking outside the tree. Ensure this behavior is intended, as it might conflict with other interactive elements on the page.